packages/mobile/src/components/audio/Audio.tsx

```diff
@@ -268,13 +273,13 @@ export const Audio = () => {
     const duration = await TrackPlayer.getDuration()
     const position = await TrackPlayer.getPosition()
 
-    if (
-      position > RECORD_LISTEN_SECONDS &&
-      !listenLoggedForTrack &&
-      isReachable
-    ) {
+    if (position > RECORD_LISTEN_SECONDS && !listenLoggedForTrack) {
       setListenLoggedForTrack(true)
-      dispatch(recordListen(trackId))
+      if (isOfflineModeEnabled && !isReachable) {
+        queue.addJob<PlayCountWorkerPayload>(PLAY_COUNTER_WORKER, { trackId })
+      } else if (isReachable) {
+        dispatch(recordListen(trackId))
+      }
```

Comment:
super nitpick, but I might put the reachable case first since it's the most common one